service/history/queue/timer_queue_processor.go

```diff
@@ -365,7 +375,7 @@ func (t *timerQueueProcessor) completeTimerLoop() {
 			return
 		case <-completeTimer.C:
 			for attempt := 0; attempt < t.config.TimerProcessorCompleteTimerFailureRetryCount(); attempt++ {
-				err := t.completeTimer()
+				err := t.completeTimer(context.Background())
```

Comment:
I wonder if this should be unbounded, this is strictly better however.

Reply:
I debated on this but leaving it as is for now to limit the changes to be tested in actual deployment